ui_framework/components/button/kui_button.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import classnames from 'classnames';
+
+import { KuiButtonLink } from './kui_button_link';
+import { KuiTooltip } from '../tooltip/kui_tooltip';
+
+export function KuiButton(props) {
+  if (props.href && props.onClick) {
+    throw new Error('Only href or onClick can be defined, not both');
+  }
+
+  if (props.hide) return null;
+
+  const classes = classnames('kuiButton', props.className);
+
+  if (props.href) {
+    return <KuiButtonLink {...props} />;
```

Comment:
Let's make a hard split between `Button` and `ButtonLink`. Can we remove this condition here, disallow the `href` prop in `Button`, and disallow `onClick` in `ButtonLink`? This way, the engineer has to specify exactly which one to use, instead of delegating the decision to the code.